### PR TITLE
feat(integrations): add confirmation modal when deleting PagerDuty services

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a confirmation modal before deleting a PagerDuty service:

Screen Shot 2020-04-06 at 12 56 59 PM

I also cleaned up some TS in the TableField component

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/forms/type.tsx [hunk 1]
```diff
@@ -108,10 +108,22 @@ type RangeType = {type: 'range'} & Omit<RangeSlider['props'], 'value'> & {
     value?: Pick<RangeSlider['props'], 'value'>;
   };
 
-type TableType = {
+export type TableType = {
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/settings/components/forms/type.tsx:111]: I couldn't find a way to use these props in the TableField component without exporting like this. Might need to do this when we convert other components as well.

File: src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx [hunk 2]
```diff
@@ -10,70 +9,53 @@ import Input from 'app/views/settings/components/forms/controls/input';
...
-    const mappedKeys = columnKeys;
+    const mappedKeys = columnKeys || [];
     const emptyValue = mappedKeys.reduce((a, v) => ({...a, [v]: null}), {id: ''});
 
     const valueIsEmpty = this.hasValue(props.value);
     const value = valueIsEmpty ? (props.value as any[]) : [];
 
     const saveChanges = (nextValue: object[]) => {
-      onChange(nextValue, []);
+      onChange?.(nextValue, []);
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx:58]: The `callIfFunction` utility is another way to do this.

File: src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx [hunk 3]
```diff
@@ -126,12 +110,26 @@ export default class TableField extends React.Component<Props> {
...
+    const renderConfirmMessage = () => {
+      return (
+        <React.Fragment>
+          <Alert type="error">
+            <span
+              dangerouslySetInnerHTML={{
+                __html: singleLineRenderer(confirmDeleteMessage || ''),
+              }}
+            />
+          </Alert>
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx:122]: Is an empty alert going to look ok in the confirm box?

Comment (scefali) [reply src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx:124]: @markstory Good point, I put in a default message

File: src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx [hunk 4]
```diff
@@ -116,7 +116,9 @@ export default class TableField extends React.Component<Props> {
           <Alert type="error">
             <span
               dangerouslySetInnerHTML={{
-                __html: singleLineRenderer(confirmDeleteMessage || ''),
+                __html: singleLineRenderer(
+                  confirmDeleteMessage || 'Are you sure you want to delete this item?'
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx:120]: Should this have a t() around it?

Comment (scefali) [reply src/sentry/static/sentry/app/views/settings/components/forms/tableField.tsx:120]: @markstory We can definitely translate the default text but not the configurable one

**Review by markstory — APPROVED** [review]

**Pull request merged**